devel/libanalysis/src/module_data_block.c

```diff
@@ -0,0 +1,84 @@
+/*
+   Copyright (C) 2016  Statoil ASA, Norway.
+
+   The file 'module_data_block.c' is part of ERT - Ensemble based Reservoir Tool.
+
+   ERT is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+   FITNESS FOR A PARTICULAR PURPOSE.
+
+   See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+   for more details.
+*/
+
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <ert/util/util.h>
+#include <ert/util/type_macros.h>
+
+#include <ert/analysis/module_data_block.h>
+
+#define MODULE_DATA_BLOCK_TYPE_ID 73217801
+static UTIL_SAFE_CAST_FUNCTION( module_data_block , MODULE_DATA_BLOCK_TYPE_ID);
+
+module_data_block_type * module_data_block_alloc(  const char * key ) {
+  module_data_block_type * module_data_block = util_malloc( sizeof * module_data_block );
+  UTIL_TYPE_ID_INIT( module_data_block , MODULE_DATA_BLOCK_TYPE_ID );
+  module_data_block->key = util_alloc_string_copy( key );
+  module_data_block->A_row_start = -1;
+  module_data_block->A_row_end = -1;
+  module_data_block->index_list  = NULL;
+  return module_data_block;
+}
+
+
+void module_data_block_set_row_start(module_data_block_type * module_data_block, const int row_start ) {
+  module_data_block->A_row_start = row_start;
+}
+
+void module_data_block_set_row_end(module_data_block_type * module_data_block,  const int row_end ) {
+  module_data_block->A_row_end = row_end;
+}
+
+void module_data_block_set_active_indices(module_data_block_type * module_data_block, const int * index_list, const int n_active ) {
```

Comment:
Is it really necessary with all these `set_xxx` functions? I would assume it should be possible to get RAII behavior with something like:
```C
module_data_block_type * module_data_block_alloc( const char * key , const int * index_list , int row_start) {
    ...
}
```
and from then on the `module_block_data` object could be immutable.